File: apps/render-android-playground/app/src/main/java/com/max_mrtnv/render_android_playground/domain/entities/Scenario.kt
package com.max_mrtnv.render_android_playground.domain.entities

import java.util.UUID

/**
 * Domain entity representing a complete UI scenario
 */
data class Scenario(
    val id: String,
    val key: String,
    val version: String,
    val buildNumber: Int,
    val mainComponent: Component,
    val components: Map<String, Component>,
    val metadata: Map<String, Any>,
    val createdAt: Long,
    val updatedAt: Long
) {
    companion object {
        fun create(json: Map<String, Any?>): Scenario? {
            val config = Config(json)
            
            val id = config.getString("id") ?: UUID.randomUUID().toString()
            val key = config.getString("key") ?: config.getString("id") ?: UUID.randomUUID().toString()
            val version = config.getString("version") ?: "1.0.0"
            val buildNumber = config.getInt("buildNumber") ?: 1
            val metadata = config.getDictionary("metadata") ?: emptyMap()
            val createdAt = config.getLong("createdAt") ?: System.currentTimeMillis()
            val updatedAt = config.getLong("updatedAt") ?: System.currentTimeMillis()
            
            val mainComponentConfig = config.getConfig("main") ?: return null
            val mainComponent = try {
                Component.create(mainComponentConfig)
            } catch (e: Exception) {
                println("Error creating main component: ${e.message}")
                return null
            }
            
            // Parse components section
            val components = mutableMapOf<String, Component>()
            val componentsSection = config.getDictionary("components")
            componentsSection?.forEach { (componentName, componentData) ->
                if (componentData is Map<*, *>) {
                    try {
                        val component = Component.create(Config(componentData))
                        components[componentName] = component
                    } catch (e: Exception) {
                        println("Error creating component $componentName: ${e.message}")
                    }
                }
            }
            
            return Scenario(id, key, version, buildNumber, mainComponent, components, metadata, createdAt, updatedAt)
        }
    }
}

File: apps/render-ios-playground/render-ios-playground/Domain/Entities/Scenario.swift
import Foundation

/// Domain entity representing a complete UI schema
public class Scenario {
    let id: String
    let key: String
    let mainComponent: Component
    let components: [String: Component]
    let version: String
    let build_number: Int
    let metadata: [String: Any]
    let createdAt: Date
    let updatedAt: Date

    private init(
        id: String,
        key: String,
        mainComponent: Component,
        components: [String: Component],
        version: String,
        build_number: Int,
        metadata: [String: Any],
        createdAt: Date,
        updatedAt: Date
    ) {
        self.id = id
        self.key = key
        self.version = version
        self.build_number = build_number
        self.mainComponent = mainComponent
        self.metadata = metadata
        self.components = components
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
    
    static func create(from json: [String: Any?]) -> Scenario? {
        let config = Config(json)

        let id = config.getString(forKey: "id") ?? UUID().uuidString
        let key = config.getString(forKey: "key") ?? config.getString(forKey: "id") ?? UUID().uuidString
        let version = config.getString(forKey: "version") ?? "1.0.0"
        let build_number = config.getInt(forKey: "build_number") ?? 1
        let metadata = config.getDictionary(forKey: "metadata") ?? [:]
        let mainComponentConfig = config.getConfig(forKey: "mainComponent")

        guard
            !mainComponentConfig.isEmpty,
            let mainComponent = try? Component.create(from: mainComponentConfig)
        else { return nil }

        // Parse components section
        var components: [String: Component] = [:]
        if let componentsSection = config.getDictionary(forKey: "components") {
            for (componentName, componentData) in componentsSection {
                if let componentData = componentData as? [String: Any?],
                   let component = try? Component.create(from: Config(componentData)) {
                    components[componentName] = component
                }
            }
        }

        // Parse timestamps
        let createdAt = config.getDate(forKey: "createdAt") ?? Date()
        let updatedAt = config.getDate(forKey: "updatedAt") ?? Date()

        return Scenario(
            id: id,
            key: key,
            mainComponent: mainComponent,
            components: components,
            version: version,
            build_number: build_number,
            metadata: metadata,
            createdAt: createdAt,
            updatedAt: updatedAt
        )
    }
}
